### PR TITLE
Update ITC & UTC with generic config's 

### PR DESCRIPTION
**Pull request by thapav**
This PR adds generic config's to ITC's & UTC's so that user can change values directly through menuconfig instead of hardcoding them.

### CODE REVIEW

File: apps/examples/testcase/ta_tc/systemio/itc/Kconfig [hunk 1]
```diff
@@ -14,20 +14,79 @@ config SYSIO_ITC_UART
...
+	int "Sysio PWM test period"
+	default 1000
+endif
+
 config SYSIO_ITC_GPIO
 	bool "Sysio GPIO TC"
 	default n
+
+if SYSIO_ITC_GPIO
+config SYSIO_ITC_GPIO1_NUM
```
Comment (Taejun-Kwon) [inline apps/examples/testcase/ta_tc/systemio/itc/Kconfig:82]: postfix 'NUM' is unnecessary I think, or 'PORT' is better option. How do you think?

Comment (sunghan-chang) [reply apps/examples/testcase/ta_tc/systemio/itc/Kconfig:82]: @thapav Could you reply Taejun's comment?

Comment (thapav) [reply apps/examples/testcase/ta_tc/systemio/itc/Kconfig:82]: Yes @Taejun-Kwon  we could remove it. Let me update it

Comment (thapav) [reply apps/examples/testcase/ta_tc/systemio/itc/Kconfig]: Modified, thanks.

**Review by sunghan-chang — APPROVED** [review]

**Pull request merged**